modules/oe_subscriptions_anonymous/tests/src/Functional/HtmlMailsTest.php

```diff
@@ -0,0 +1,250 @@
+      "Click the following link to confirm your subscription: {$confirm_link['text']} " .
+      "If you no longer wish to subscribe, click on the link bellow: {$cancel_link['text']} " .
+      "If you didn't subscribe to these updates or you're not sure why you received this e-mail, you can delete it. " .
+      "You will not be subscribed if you don't click on the confirmation link above.",
+      $wrapper->text());
+  }
+
+  /**
+   * Asserts HTML in body for Subscriptions page mail.
+   *
+   * @param array $site_link
+   *   The link to the site.
+   * @param array $subscriptions_link
+   *   The link to the subscriptions page.
+   * @param string $mail_body
+   *   The mail body.
+   */
+  protected function assertSubscriptionsMailHtml(array $site_link, array $subscriptions_link, string $mail_body): void {
+    $crawler = new Crawler($mail_body);
+    $wrapper = $crawler->filter('.email-sub-type-user-subscriptions-access div.clearfix');
+    $assert_session = $this->assertSession();
+
+    // Check that the links are in the text.
+    [$subscriptions_url] = $this->assertLinks([$site_link, $subscriptions_link], $wrapper);
+
+    // Assert subscription link.
+    $this->drupalGet($subscriptions_url);
+    $assert_session->titleEquals('Manage your subscriptions | Drupal');
+    $assert_session->elementExists('css', 'table.user-subscriptions');
+
+    // Check break lines.
+    $br = $wrapper->filter('br');
+    $this->assertCount(2, $br);
+    $this->assertStringContainsString($site_link['text'] . '</a>.' . $br->html(), $mail_body);
+    $this->assertStringContainsString($subscriptions_link['text'] . '</a>' . $br->html(), $mail_body);
+
+    // Check the text.
+    $this->assertEquals(
+      "You are receiving this e-mail because you requested access to your subscriptions page on {$site_link['text']}. " .
+      "Click the following link to access your subscriptions page: {$subscriptions_link['text']} " .
+      "If you didn't request access to your subscriptions page or you're not sure why you received this e-mail, you can delete it.",
+      $wrapper->text());
+  }
+
+  /**
+   * Asserts links skipping those without URL key value.
+   *
+   * @param array[] $expected_links
+   *   A list links to check.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler where perform the checks.
+   *
+   * @return array
+   *   The skipped urls.
+   */
+  protected function assertLinks(array $expected_links, Crawler $crawler): array {
+    $skipped_urls = [];
+
+    foreach ($expected_links as $expected_link) {
+      $link = $crawler->filterXPath("//a[contains(text(),'{$expected_link['text']}')]");
+      $this->assertCount(1, $link);
+      if (isset($expected_link['url'])) {
+        $this->assertEquals($expected_link['url'], $link->attr('href'));
+        continue;
+      }
+      $skipped_urls[] = $link->attr('href');
+    }
+
+    return $skipped_urls;
```

Comment:
Can't we just call this "text_only_urls"? Urls of which we checked only the text.

Reply:
Renamed